src/test/java/org/example/servletsHomework/servlet/TestTextServlets/TextsServletTest.java

```diff
@@ -0,0 +1,203 @@
+import org.example.servletsHomework.storage.TokensAndUserStorage;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.*;
+import java.util.List;
+
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+class TextsServletTest {
+    private TextsServlet textsServlet;
+    private MockHttpServletRequest request;
+    private MockHttpServletResponse response;
+    private MockPrintWriter writer;
+
+
+    @BeforeEach
+    void setUp() {
+
+        TextDao textDao = new MockTextDao2();
+        TextService textService = new TextService(textDao);
+        TokensAndUserStorage tokensAndUserStorage = new TokensAndUserStorage();
+        textsServlet = new TextsServlet(new ObjectMapper(), new MockIdGenerator(1L), textService, tokensAndUserStorage);
+        writer = new MockPrintWriter(new StringWriter());
+
+        request = new MockHttpServletRequest();
+        response = new MockHttpServletResponse(writer);
+
+        String tokenUserOne = "validToken1";
+        String tokenUserTwo = "validToken2";
+
+        User userOne = new User("testUsername1", "testPassword1", 1L, System.currentTimeMillis());
+        User userTwo = new User("testUsername2", "testPassword2", 2L, System.currentTimeMillis());
+
+        tokensAndUserStorage.addToken(tokenUserOne, userOne);
+        tokensAndUserStorage.addToken(tokenUserTwo, userTwo);
+
+        textService.addText(1L, new Texts(1L, "text1"));
+        textService.addText(2L, new Texts(1L, "text3"));
+    }
+
+    @Test
+    void shouldReturnAllTexts() {
+
+        request.setHeaders("token", "validToken1");
+        request.setMethod("GET");
+        request.setPathInfo("/");
+
+        textsServlet.handleGet(request, response);
+
+        Integer expectedStatus = 200;
+        String expectedOutput = "[[Texts{id=1, value='Test'}]]";
+
+        assertEquals(expectedOutput, writer.writtenStrings.toString());
+        assertEquals(expectedStatus, response.getStatus());
+    }
+
+    @Test
+    void shouldReturnTextById() {
+
+        request.setHeaders("token", "validToken1");
+        request.setMethod("GET");
+        request.setPathInfo("/1");
+
+        textsServlet.handleGet(request, response);
+
+        Integer expectedStatus = 200;
+        String expectedOutput = "[Text with id 1: TestText]";
+
+        assertEquals(expectedOutput, writer.writtenStrings.toString());
+        assertEquals(expectedStatus, response.getStatus());
+    }
+
+
+    @Test
+    void shouldThrowBadRequestExceptionWhenRequestWithInvalidId() {
+
+        request.setHeaders("token", "validToken1");
+        request.setMethod("GET");
+
+        List<String> invalidPaths = List.of("/a", "/ ", "/!", "%", "1/2/3/");
+
+        for (String path : invalidPaths) {
+            request.setPathInfo(path);
+            BadRequestException exception = assertThrows(BadRequestException.class, () -> textsServlet.handleGet(request, response));
```

Comment:
Why for get text by id you verify response code but for this one you expecting error ? What if error is thorwn  but response code is still 200 or 500 ? Then your Serlet works correct ?